### PR TITLE
Make min duration configurable for slowest tests

### PR DESCRIPTION
**Pull request by ssbarnea**
Enables users to use custom values for the threshold used to determine slow tests.

### CODE REVIEW

**Comment by ssbarnea** [comment]
> I prefer `--durations-threshold` over `--durations-min`, I think it is more understandable. WDYT?
> 
> Also, do you foresee this being used more at the project level, or the single `pytest` invocation level? If it is something that makes more sense to set project-wide, then an INI would make more sense than a command line flag.

I prefer min as is easier to type, not really confusing. Regarding config, I partially agree because the only reason I did not do it is because current implementation make is harder to do it. I would prefer spending time on replacing the config engine with something that maps CLI arguments to config file items and env vars naturally, without extra code, so when someone adds a new configurable, it will be exposed to all places automatically.

We can still configure options in config files using addopts, so not a deal breaker.

File: changelog/7667.feature.rst
Comment (webknjaz) [inline changelog/7667.feature.rst:1]: It's RST, double backticks should be used

Comment (ssbarnea) [reply changelog/7667.feature.rst]: Done.

File: src/_pytest/runner.py
Comment (webknjaz) [inline src/_pytest/runner.py:91]: Is "%gs" valid?

Comment (webknjaz) [reply src/_pytest/runner.py:91]:
```suggestion
                "(%s durations < %ss hidden.  Use -vv to show these durations.)"
```
Typo?

File: src/_pytest/runner.py
Comment (webknjaz) [inline src/_pytest/runner.py:92]: FWIW it's probably easier to just use an f-string

File: src/_pytest/runner.py
Comment (webknjaz) [inline src/_pytest/runner.py:61]:
```suggestion
        help="Minimal duration in seconds for inclusion in slowest list.",
```
Default is probably already appended by argparse, right?

**Comment by webknjaz** [comment]
@bluetech I think "threshold" may mean min/max depending on the PoV/context. So I'd avoid using it.

File: src/_pytest/runner.py
Comment (ssbarnea) [reply src/_pytest/runner.py:91]: `%gs` is valid and used there on purpose, in order to print variable precision, aka human numbers. "1.000" will render as "1" but "1.010" as "1.01".

File: src/_pytest/runner.py
Comment (ssbarnea) [reply src/_pytest/runner.py:92]: pytest did not switch to `require_python>=3.6` yet, but I would support dropping py35.

File: changelog/7667.feature.rst
Comment (nicoddemus) [reply changelog/7667.feature.rst]: > It's RST, double backticks should be used

You are correct, but we changed our sphinx configuration to consider single ticks as monospaced font, so either way is OK for the docs/changelog entries.

**Comment by nicoddemus** [comment]
> @bluetech I think "threshold" may mean min/max depending on the PoV/context. So I'd avoid using it.

Good point.

@bluetech would you like to take a second look or we can merge this?

File: src/_pytest/runner.py
Comment (ssbarnea) [reply src/_pytest/runner.py:61]: It is not, that is why I added it explicitly. I am not sure why but I know is not printed.

**Review by bluetech — APPROVED** [review]
Yes, LGTM.

Regarding the name, a few thoughts, "take it or leave it":

Maybe it's just me but I do find the name `--durations-min` and the explanation not entirely obvious, that is it takes me a moment to understand what it does. For me, the word "minimum" indicates a specific value from a set, not a threshold/limit, but maybe it's just me.

I think a more direct help text would be "Only include durations longer than this value (in seconds)". Which suggests other names for the flag, maybe `--durations-above=0.005`.

Or we can think about the other way and say `--durations-hide=0.005` with a help text "Hide durations below this value (in seconds)". This corresponds nicely to the "20 durations < 0.005s hidden" message.

**Comment by webknjaz** [comment]
Good. I like `--durations-above`. How about `--durations-at-least`?

**Comment by ssbarnea** [comment]
I cannot refrain to mention that two weeks after creating this PR is still not merged, main reason being our inability to agree on naming.

IMHO, it is often worse not to make a decision than making a bad one. In my particular case there is not real pressure to get this new feature merged but I can see how the same kind of issue may deter others from becoming active contributors.

Any core can edit this PR to change the name. I would not mind.

> ...me thinking that maybe someone wants to modify argparse to make it more flexible and allow `--durations-(min|threshold|above|at-least|hide|omg|wtf|...)` ;)

**Comment by graingert** [comment]
If it's only the name, I think we should merge this, then change the name in a follow-up

**Review by graingert — APPROVED** [review]

**Comment by ssbarnea** [comment]
Time to pray for someone to press the merge button :D

**Comment by nicoddemus** [comment]
> In my particular case there is not real pressure to get this new feature merged but I can see how the same kind of issue may deter others from becoming active contributors.

I understand how one can feel like that, but we're really trying our best: we work on pytest on our own free time, and time is often short and things come up which sometimes take precedence. Also note that this PR doesn't need to be rushed as it only affects the next minor release anyway, which is something we try to do around monthly.

> Time to pray for someone to press the merge button :D

Indeed, let's get this in. I'm fine with the current name, but please if somebody else prefers to change it open a new PR and it will be welcome! 😁

**Pull request merged**
